staging/src/k8s.io/apiserver/pkg/registry/rest/meta.go

```diff
@@ -28,7 +28,10 @@ import (
 func FillObjectMetaSystemFields(meta metav1.Object) {
 	meta.SetCreationTimestamp(metav1.Now())
 	meta.SetUID(uuid.NewUUID())
-	meta.SetSelfLink("")
+
+	if len(meta.GetSelfLink()) > 0 {
+		meta.SetSelfLink("")
```

Comment:
@liggitt ptal if this is need.

I don't think this hurts though.

Reply:
shouldn't be needed (PrepareObjectForStorage will do the same thing unconditionally). let's keep this PR scoped just to the things needed to fix the resourceVersion issue. fixing the unstructured accessor in the follow-up PR would mean we wouldn't care about this